lib/dns/resource.js

```diff
@@ -178,19 +178,21 @@ class Resource extends Struct {
 
   toGlue(name) {
     const additional = [];
+    let synth = false;
 
     for (const record of this.records) {
       switch (record.type) {
-        case hsTypes.GLUE4:
-        case hsTypes.GLUE6:
         case hsTypes.SYNTH4:
         case hsTypes.SYNTH6:
+          synth = true;
+        case hsTypes.GLUE4:
+        case hsTypes.GLUE6:
           break;
         default:
           continue;
       }
 
-      if (!util.isSubdomain(name, record.ns))
+      if (!util.isSubdomain(name, record.ns) && !synth)
```

Comment:
TODO: does this also have to happen here?

https://github.com/pinheadmz/hsd/blob/d09354fff66e1b730a4673a6f9497683eb56b178/lib/dns/resource.js#L259-L273